src/main/java/fr/sncf/osrd/railjson/parser/RailJSONParser.java

```diff
@@ -174,7 +174,10 @@ public static Infra parse(RJSInfra railJSON) throws InvalidInfraException {
                         rjsSignal.id,
                         expr,
                         rjsSignal.applicableDirection,
-                        rjsSignal.sightDistance
+                        rjsSignal.sightDistance,
+                        ((rjsSignal.linkedDetector != null && !rjsSignal.linkedDetector.id.equals(""))
+                                ? (Detector) waypointsMap.get(rjsSignal.linkedDetector.id)
+                                : null)
```

Comment:
That is an awkward way to write this, move it to a variable before. Isn't there a way to use `getOrDefault` here?